### PR TITLE
[ADF-4586] - fix preselecting value for DropdownCloud

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-4586

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/form/components/form-cloud.component.ts [hunk 1]
```diff
@@ -78,12 +87,15 @@ export class FormCloudComponent extends FormBaseComponent implements OnChanges {
...
                 protected formService: FormService,
                 private notificationService: NotificationService,
+                private formRenderingService: FormRenderingService,
                 protected visibilityService: WidgetVisibilityService) {
         super();
 
         this.formService.formContentClicked.subscribe((content: any) => {
             this.formContentClicked.emit(content);
         });
+        this.formRenderingService.setComponentTypeResolver('upload', () => UploadCloudWidgetComponent, true);
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/form/components/form-cloud.component.ts]: @SilviuCPopa @mauriziovitale Every time this component is created, it rewrites the `upload` and `dropdown` entries for the entire application. Meaning opening the Form after closing this one will still mean using cloud versions. If you are not fine with that - make a copy of the `FormRenderingService` by declaring it in the `providers` section of this component.

Comment (SilviuCPopa) [reply lib/process-services-cloud/src/lib/form/components/form-cloud.component.ts]: @DenysVuika added FormRenderingService as provider. Thanks

File: lib/process-services-cloud/src/lib/form/components/form-cloud.component.ts [hunk 2]
```diff
@@ -78,12 +87,15 @@ export class FormCloudComponent extends FormBaseComponent implements OnChanges {
     constructor(protected formCloudService: FormCloudService,
                 protected formService: FormService,
                 private notificationService: NotificationService,
+                private formRenderingService: FormRenderingService,
                 protected visibilityService: WidgetVisibilityService) {
         super();
 
         this.formService.formContentClicked.subscribe((content: any) => {
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/form/components/form-cloud.component.ts]: Memory leak, missing unsubscribe, so the service will keep in memory all instances 😭

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts [hunk 3]
```diff
@@ -67,7 +68,20 @@ export class DropdownCloudWidgetComponent extends WidgetComponent implements OnI
         });
      }
 
-    getOptionValue(option: FormFieldOption, fieldValue: string): string {
+     private setOptionValues(result: any) {
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts]: missing type information
```suggestion
     private setOptionValues(result: { id: string; name: string }) {
```

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts [hunk 4]
```diff
@@ -67,7 +74,20 @@ export class DropdownCloudWidgetComponent extends WidgetComponent implements OnI
         });
      }
 
-    getOptionValue(option: FormFieldOption, fieldValue: string): string {
+     private setOptionValues(result: FormFieldOption[] ) {
+        this.field.options = result.map( (value: any) => {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts]: the value is it any?

**Pull request merged**